src/CashuWallet.ts

```diff
@@ -474,22 +473,43 @@ class CashuWallet {
 		}
 		const sendBlindedMessages = this.createRandomBlindedMessages(amount, preference, counter);
 
-		// join keepBlindedMessages and sendBlindedMessages
+		const convertedKeepMessages = keepBlindedMessages.blindedMessages.map((_, i) => ({
+			msg: keepBlindedMessages.blindedMessages[i],
+			secret: keepBlindedMessages.secrets[i],
+			rs: keepBlindedMessages.rs[i],
+			amount: keepBlindedMessages.amounts[i],
+			keep: true
+		}));
+
+		const convertedSendMessages = sendBlindedMessages.blindedMessages.map((_, i) => ({
+			msg: sendBlindedMessages.blindedMessages[i],
+			secret: sendBlindedMessages.secrets[i],
+			rs: sendBlindedMessages.rs[i],
+			amount: sendBlindedMessages.amounts[i],
+			keep: false
+		}));
+		const keepVector: Array<boolean> = [];
 		const blindedMessages: BlindedTransaction = {
```

Comment:
let's rename this variable to blindedTransaction. `blindedMessages` seems kinda confusing. What do you think?

Reply:
Oh yes absolutely. Will do